File: PakingLotDemo/src/factories/VehicleFactory.java
package factories;

import enums.VehicleSize;
import model.Bike;
import model.Car;
import model.Truck;
import model.Vehicle;

/**
 * Factory for creating vehicles by size (SMALL→Bike, MEDIUM→Car, LARGE→Truck).
 */
public class VehicleFactory {

    /**
     * Creates a vehicle of the appropriate type for the given size.
     *
     * @param vehicleNumber the vehicle identifier (e.g. license plate)
     * @param vehicleSize   the size (SMALL, MEDIUM, LARGE)
     * @return the created vehicle
     * @throws IllegalArgumentException if vehicleSize is unknown
     */
    public static Vehicle createVehicle(String vehicleNumber, VehicleSize vehicleSize) {
        return switch (vehicleSize) {
            case LARGE -> new Truck(vehicleNumber, vehicleSize);
            case MEDIUM -> new Car(vehicleNumber, vehicleSize);
            case SMALL -> new Bike(vehicleNumber, vehicleSize);
            default -> throw new IllegalArgumentException("Unknown Vehicle: " + vehicleSize);
        };
    }
}

File: problems/python/ParkingLotSystem/src/factories/vehicle_factory.py
"""Factory for creating vehicles by size."""
from enums.vehicle_size import VehicleSize
from model.vehicle import Vehicle
from model.bike import Bike
from model.car import Car
from model.truck import Truck


class VehicleFactory:
    @staticmethod
    def create_vehicle(vehicle_number: str, vehicle_size: VehicleSize) -> Vehicle:
        if vehicle_size == VehicleSize.SMALL:
            return Bike(vehicle_number, vehicle_size)
        if vehicle_size == VehicleSize.MEDIUM:
            return Car(vehicle_number, vehicle_size)
        if vehicle_size == VehicleSize.LARGE:
            return Truck(vehicle_number, vehicle_size)
        raise ValueError(f"Unknown Vehicle: {vehicle_size}")
